### PR TITLE
Integration Tests - First look

### PR DESCRIPTION
**Pull request by akhera99**
Still working through some settings shenanigans

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -118,26 +108,45 @@
...
-            "stopOnEntry": false,
             "sourceMaps": true,
             "outFiles": [
                 "${workspaceRoot}/dist/*.js",
                 "${workspaceRoot}/out/test/**/*.js"
             ],
             "preLaunchTask": "buildDev"
         },
+        {
+            "name": "Blue: Launch slnWithCsproj Workspace Tests",
```
Comment (dibarbet) [inline .vscode/launch.json:127]: lets keep 'blue' out of the code.  Good names could be 'LSP tools host', 'LSP', 'Roslyn Server', etc.  Same for the rest of the file / folder names (not commenting them individually)

Comment (dibarbet) [reply .vscode/launch.json:127]: could also consider having the new tests just be 'integrationTests' and renaming the O# folder/tests to 'omnisharpIntegrationTests'.  That I think might be the most clear

File: bluetest/integrationTestsBlue/integrationHelpersBlue.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import { CSharpExtensionExports } from '../../src/csharpExtensionExports';
+
+export async function restartRoslynLanguageServer(): Promise<void> {
+    const csharpExtension = vscode.extensions.getExtension<CSharpExtensionExports>('ms-dotnettools.csharp');
+    if (!csharpExtension) {
+        throw new Error('Failed to find installation of ms-dotnettools.csharp');
+    }
+
+    if (!csharpExtension.isActive) {
+        await activateCSharpExtension();
```
Comment (dibarbet) [inline bluetest/integrationTestsBlue/integrationHelpersBlue.ts:17]: This won't necessarily restart the server.  If activation is complete and the runtime extension is installed then nothing will happen

File: bluetest/integrationTestsBlue/integrationHelpersBlue.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    // Explicitly await the extension activation even if completed so that we capture any errors it threw during activation.
+    await csharpExtension.activate();
+
+    await csharpExtension.exports.initializationFinished();
+    console.log('ms-dotnettools.csharp activated');
+}
+
+export function isRazorWorkspace(workspace: typeof vscode.workspace) {
```
Comment (dibarbet) [inline bluetest/integrationTestsBlue/integrationHelpersBlue.ts:43]: delete everything not being used in this PR

File: bluetest/integrationTestsBlue/integrationHelpersBlue.ts (hunk 2 above)
Comment (dibarbet) [reply bluetest/integrationTestsBlue/integrationHelpersBlue.ts:17]: You probably want invoke the restart server or reload command explicitly

File: bluetest/testRunner.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+// Linux: prevent a weird NPE when mocha on Linux requires the window size from the TTY
+// Since we are not running in a tty environment, we just implementt he method statically
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+tty.getWindowSize = function () {
+    return [80, 75];
+};
+
+function setupLogging() {
+    if (process.env.CODE_EXTENSIONS_PATH && process.env.OSVC_SUITE) {
```
Comment (dibarbet) [inline bluetest/testRunner.ts:22]: what is osvc_suite, do we need it here?

File: bluetest/testRunner.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        const logDirPath = path.join(process.env.CODE_EXTENSIONS_PATH, './.logs');
+
+        if (!fs.existsSync(logDirPath)) {
+            fs.mkdirSync(logDirPath);
+        }
+
+        const logFilePath = path.join(logDirPath, `${process.env.OSVC_SUITE}.log`);
+
+        SubscribeToAllLoggers((message) => fs.appendFileSync(logFilePath, message));
```
Comment (dibarbet) [inline bluetest/testRunner.ts:31]: we don't use loggers in the new tools host, don't think we necessarily need this part.

File: omnisharptest/integrationTests/index.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+/*---------------------------------------------------------------------------------------------
```
Comment (akhera99) [inline omnisharptest/integrationTests/index.ts:1]: rename, no file changes

File: omnisharptest/integrationTests/integrationHelpers.ts [hunk 7]
```diff
@@ -0,0 +1,104 @@
+/*---------------------------------------------------------------------------------------------
```
Comment (akhera99) [inline omnisharptest/integrationTests/integrationHelpers.ts:1]: rename, no file changes

File: omnisharptest/testRunner.ts [hunk 8]
```diff
@@ -0,0 +1,66 @@
+/*---------------------------------------------------------------------------------------------
```
Comment (akhera99) [inline omnisharptest/testRunner.ts:1]: rename only, no file changes

File: lsptoolshosttest/lspToolsHostIntegrationTests/index.ts [hunk 9]
```diff
@@ -0,0 +1,17 @@
+/*---------------------------------------------------------------------------------------------
```
Comment (dibarbet) [inline lsptoolshosttest/lspToolsHostIntegrationTests/index.ts:1]: the plan is to rename the old tests and these in a followup PR right?

**Review by dibarbet — APPROVED** [review]
So for the followup prs
1.  Renaming the tests
2.  Enable the tests in CI - ideally we would have 2 different steps for O# tests and these tests.  And hopefully we could enable these tests to run on linux, mac and and windows.
     a.  as part of the CI changes we should also upload the output window logs (we can talk offline on that)

**Comment by akhera99** [comment]
> So for the followup prs
> 
> 1. Renaming the tests
> 2. Enable the tests in CI - ideally we would have 2 different steps for O# tests and these tests.  And hopefully we could enable these tests to run on linux, mac and and windows.
>    a.  as part of the CI changes we should also upload the output window logs (we can talk offline on that)

Yep!

**Pull request merged**